src/Vector3.h

```diff
@@ -235,6 +235,12 @@ class Vector3
 		return *this;
 	}
 
+	/** Provides a hash of a vector's contents */
+	size_t operator()(const Vector3<T> & a_Vector) const
+	{
+		return ((std::hash<T>()(a_Vector.x) ^ (std::hash<T>()(a_Vector.y) << 1)) ^ std::hash<T>()(a_Vector.z));
```

Comment:
This has function is really bad for our dataset, as there will be lots of collisions within a chunk sized area. @SafwatHalaby had a much better hash function for our needs.